rascaline/src/systems/neighbors.rs

```diff
@@ -250,10 +250,28 @@ impl CellList {
                                 let shift = CellShift(cell_shift) + atom_i.shift - atom_j.shift;
                                 let shift_is_zero = shift[0] == 0 && shift[1] == 0 && shift[2] == 0;
 
-                                if atom_i.index == atom_j.index && shift_is_zero {
-                                    // only create pair with the same atom twice
-                                    // if the pair spans more than one unit cell
-                                    continue;
+                                if atom_i.index == atom_j.index {
+                                    if shift_is_zero {
+                                        // only create pair with the same atom twice
+                                        // if the pair spans more than one unit cell
+                                        continue;
+                                    }
+
+                                    if (shift[0] + shift[1] + shift[2] < 0) ||
+                                        (
+                                            (shift[0] + shift[1] + shift[2] == 0) &&
+                                            (shift[2] < 0 || (shift[2] == 0 && shift[1] < 0))
+                                        )
+                                    {
+                                        // When creating pairs between an atom
```

Comment:
I understand what is going on here but maybe a picture would help. But I agree and accept, if this is too much work.

Reply:
does this work?

```
──────┼─────┼─────┼─────┼─────┼
     ╭╯    ╭╯    ╭╯    ╭╯    ╭╯
     │     │     │   × │     │
     │     │     │  ╱  │     │
    ╭╯    ╭╯    ╭╯ ╱  <- first pair
────┼─────┼─────┼─────┼─────┼
   ╭╯    ╭╯   ╱╭╯    ╭╯    ╭╯
   │     │   × │     │     │
   │     │  ╱  │     │     │
  ╭╯    ╭╯ ╱  <- second pair, redundant
──┼─────┼─────┼─────┼─────┼
 ╭╯    ╭╯    ╭╯    ╭╯    ╭╯
 │    ╱│     │     │     │
 │   × │     │     │     │
╭╯    ╭╯    ╭╯    ╭╯    ╭╯
┼─────┼─────┼─────┼─────┼
```

Reply:
We looked at the code very carefully again, and I think we both understand it now! I might do a diagram of the 2D version of it, but trying to represent the 3D version in ASCII is a bit clunky

Reply:
I think I understand what the hard part is in the understanding. my reasoning is:
- for two same-atom pairs characterised by (h,j,k) and (-h,-j,-k), one of them has to be ignored. how do you choose?
- first try to divide space along the h+j+k=0 plane.
- if it doesn't work, divide by the k=0 plane
- if it still doesn't work, divide by the j=0 plane.
- if the two pairs cannot be separated by any of those planes, then they are both at (h=0,j=0,k=0), which isn't possible

Reply:
Yes, that was the issue. I split the condition in two and added a small illustration of the 2D case